src/Reservation.php

```diff
@@ -181,6 +182,94 @@ public function prepareInputForAdd($input)
         return parent::prepareInputForAdd($input);
     }
 
+    public static function handleAddForm(array $input)
+    {
+        $rr = new self();
+        $dates_to_add = [];
+
+        if (isset($input['resa']["end"])) {
+            $dates_to_add[$input['resa']["begin"]] = $input['resa']["end"];
+
+            if (
+                isset($input['periodicity'])
+                && isset($input['periodicity']['type'])
+                && !empty($input['periodicity']['type'])
+            ) {
+                $dates_to_add += Reservation::computePeriodicities(
+                    $input['resa']["begin"],
+                    $input['resa']["end"],
+                    $input['periodicity']
+                );
+            }
+        }
+
+        if (isset($input['items']) && count($input['items']) && isset($input['users_id'])) {
+            foreach ($input['items'] as $reservationitems_id) {
+                $data = [
+                    'begin' => $input['resa']["begin"] ?? $input['begin'],
+                    'end' => $input['resa']["end"] ?? $input['end'],
+                    'reservationitems_id' => $reservationitems_id,
+                    'comment' => $input['comment'],
+                    'users_id' => $input['users_id'],
+                ];
+
+                if (!empty($dates_to_add)) {
+                    ksort($dates_to_add);
+                    $data['group'] = (count($dates_to_add) > 1) ? $rr->getUniqueGroupFor($reservationitems_id) : 0;
```

Comment:
No need to use a temporary array for just one value.
```suggestion
                    $group = (count($dates_to_add) > 1) ? $rr->getUniqueGroupFor($reservationitems_id) : 0;
```